### PR TITLE
Register func to doc file option addition

### PR DESCRIPTION
**Pull request by Ferrany1**
**Describe the PR**
Register func to doc file option addition

**Relation issue**
Replacing init in doc.go as a Register method
https://github.com/swaggo/swag/issues/1113

### CODE REVIEW

**Comment by Ferrany1** [comment]
@ubogdan Do I need to cover this case?

**Comment by ubogdan** [comment]
Yes, please.

**Comment by Ferrany1** [comment]
@ubogdan done

File: swagger.go [hunk 1]
```diff
@@ -19,22 +19,28 @@ type Swagger interface {
...
 }
 
+// Documentation is an interface to register swagger document.
+type Documentation interface {
+	Register() (string, error)
+}
+
 // Register registers swagger for given name.
-func Register(name string, swagger Swagger) {
+func Register(name string, swagger Swagger) error {
```
Comment (ubogdan) [inline swagger.go:28]: Changing the function signature introduces a breaking change.

File: gen/gen_test.go [hunk 2]
```diff
@@ -101,13 +101,17 @@ func TestGen_BuildInstanceName(t *testing.T) {
...
 	config.InstanceName = "custom"
+	config.DocUserRegister = true
 	goSourceFile = filepath.Join(config.OutputDir, config.InstanceName+"_"+"docs.go")
 	assert.NoError(t, New().Build(config))
 	expectedCode, err = ioutil.ReadFile(goSourceFile)
 	if err != nil {
 		require.NoError(t, err)
 	}
-	if !strings.Contains(string(expectedCode), "swag.Register(\"custom\", &s{})") {
+	println(string(expectedCode))
```
Comment (ubogdan) [inline gen/gen_test.go:111]: this has to be cleaned

**Review by ubogdan — CHANGES_REQUESTED** [review]

File: swagger.go (hunk 1 above)
Comment (Ferrany1) [reply swagger.go:28]: Isn't it better to allow user to handle error by himself instead of panicking?

**Pull request closed without merging**